pymeasure/instruments/agilent/agilent34450A.py

```diff
@@ -0,0 +1,571 @@
+            else:
+                self.frequency_voltage_range = measured_from_range
+        elif measured_from == "current_ac":
+            self.mode = "current frequency"
+            if measured_from_range == "AUTO":
+                self.frequency_current_auto_range = True
+            else:
+                self.frequency_current_range = measured_from_range
+        else:
+            raise ValueError('Incorrect value for measured_from parameter. Use '
+                             '"voltage_ac" or "current_ac".')
+        self.frequency_aperture = aperture
+
+    def configure_temperature(self):
+        """ Configures the instrument to measure temperature.
+        """
+        self.mode = 'temperature'
+
+    def configure_diode(self):
+        """ Configures the instrument to measure diode voltage.
+        """
+        self.mode = 'diode'
+
+    def configure_capacitance(self, capacitance_range="AUTO"):
+        """ Configures the instrument to measure capacitance.
+
+        :param capacitance_range: A capacitance in Farads to set the capacitance range, can be
+                                    1E-9, 10E-9, 100E-9, 1E-6, 10E-6, 100E-6, 1E-3, 10E-3,
+                                    as well as "MIN", "MAX", "DEF" (1E-6), or "AUTO".
+        """
+        self.mode = 'capacitance'
+        if capacitance_range == "AUTO":
+            self.capacitance_auto_range = True
+        else:
+            self.capacitance_range = capacitance_range
+
+    def configure_continuity(self):
+        """ Configures the instrument to measure continuity.
+        """
+        self.mode = 'continuity'
+
+    def beep(self):
+        """ Sounds a system beep.
+        """
+        self.write(":SYST:BEEP")
+
+    def _conf_parser(self, conf_values):
+        """
+        Parse the string of configuration parameters read from Agilent34450A with
+        command ":configure?" and returns a list of parameters.
+
+        Use cases:
+
+        ['"CURR +1.000000E-01', '+1.500000E-06"']   ** Obtained from Instrument.measurement or Instrument.control
+        '"CURR +1.000000E-01,+1.500000E-06"'        ** Obtained from Instrument.ask
+
+        becomes
+
+        ["CURR", +1000000E-01, +1.500000E-06]
+        """
+        # If not already one string, get one string
+
+        if isinstance(conf_values, list):
+            one_long_string = ', '.join(map(str, conf_values))
+        else:
+            one_long_string = conf_values
+
+        # Split string in elements
+        list_of_elements = re.split('["\s,]', one_long_string)
+
+        # Eliminate empty string elements
+        list_without_empty_elements = list(filter(lambda v: v != '', list_of_elements))
+
+        # Convert numbers from str to float, where applicable
+        for i, v in enumerate(list_without_empty_elements):
+            try:
+                list_without_empty_elements[i] = float(v)
+            except ValueError as e:
+                print(e)
```

Comment:
that should probably use the `log` you define above, instead of `print`

Reply:
Agreed, modified in new commit e0e1a7a.